### ISSUE TITLE
Trailing white space broken on Windows

### ISSUE THREAD

**Issue by ljani** [spf13/spf13-vim#340]
The commit https://github.com/spf13/spf13-vim/commit/987f6dfbb0b8e8ef481a0d29735bd6e5139b546b changed the default trailing whitespace character to `•`, which apparently doesn't work properly on Windows 7 x64, Vim 7.3 `+multi_byte`, Consolas font and `encoding=UTF-8`. It displays fine if I open the file, but in the GUI it's displayed as `¿`.

Some screenshots:
![Screenshot](http://i.imgur.com/DqgSsUt.png)
![Screenshot](http://i.imgur.com/VG0yr67.png)

**Comment by spf13** [comment]
Thanks for the feedback. Do you have a suggestion for a character that does
work on windows to replace it with?

Steve Francia
spf13.com
@spf13

On Apr 11, 2013, at 12:37 PM, Jani notifications@github.com wrote:

The commit 987f6dfhttps://github.com/spf13/spf13-vim/commit/987f6dfbb0b8e8ef481a0d29735bd6e5139b546bchanged
the default trailing whitespace character to
•, which apparently doesn't work properly on Windows 7x 64, Vim 7.3
+multi_byte and encoding=UTF-8. It displays fine if I open the file, but in
the GUI it's displayed as ¿.

Some screenshots:
[image: Screenshot]https://a248.e.akamai.net/camo.github.com/d98c83164c120e35bc8178e7fcf07160fc1430e4/687474703a2f2f692e696d6775722e636f6d2f447167537355742e706e67
[image: Screenshot]https://a248.e.akamai.net/camo.github.com/c2f7640e976f19dc09b53f7ae36ea8cdf4e3b63c/687474703a2f2f692e696d6775722e636f6d2f564730797236372e706e67

—
Reply to this email directly or view it on
GitHubhttps://github.com/spf13/spf13-vim/issues/340
.

**Comment by ljani** [comment]
Actually, the problem is not the char.. Weird. If I copy-pasted the `set listchars=tab:›\ ,trail:•,extends:#,nbsp:.` directly to my `.vimrc.local` from `.vimrc` it works just fine. I checked the files and they both encode `•` as `0xE2 0x80 0xA2`. There are no utf-8 BOMs and vim's statusbar claims that both files are encoded as utf-8. I'll try to investigate this a little more tomorrow.

**Comment by ljani** [comment]
Okay. If I try `set listchars=tab:›\ ,trail:•,extends:#,nbsp:.` before `set encoding=utf-8` I get an error saying: `Invalid argument: listchars=tab:ansiencodedstuffhere`. I wonder why vim fails silently if `listchars` is being set from spf13's `.vimrc`. Anyway, `set encoding=utf-8` before `listchars` should fix the problem.

`set encoding=utf-8` also fixes other problems too as vim defaults to ANSI on Windows 7 and 8 at least. Such as this:
![Screenshot](http://i.imgur.com/eHbIf1h.png)

EDIT: my whole encoding block in case you're interested adding it:

```
if has("multi_byte")
  if &termencoding == ""
    let &termencoding = &encoding
  endif
  set encoding=utf-8
  setglobal fileencoding=utf-8
  set fileencodings=utf-8,iso-8859-15,ucs-bom
endif
```

**Comment by spf13** [comment]
Wanna send me a pull request?

## 

Steve Francia
http://stevefrancia.com
http://spf13.com
http://twitter.com/spf13

**Comment by ljani** [comment]
If I only had a working solution, then sure :) For some reason I must always define `listchars` in my own `.vimrc.local` file, no matter if I introduce encodings in `.vimrc`.

Sure the encoding stuff fixes the statusline problems and makes the `listchars` error go away when it's defined in `.vimrc.local`, but I'd see the `listchars` "location" problem needs a solution too before making a pull request.

**Comment by ljani** [comment]
This is fixed by the pull request as setting the encoding in `.vimrc.bundles` behaves somewhat differently than setting the encoding in `.vimrc`. Also, setting the encoding in `.vimrc.bundles` kills two birds with one stone, because it fixes the installer encoding problems too as the installer only loads `.vimrc.bundles` and not the full `.vimrc` stack.

EDIT: also powerline (obviously) works properly after the commit.
EDIT2: so, yeah, kinda figured this out by accident as I was fixing the installer.

**Comment by ljani** [comment]
Fixed by #394 / fbe03d9
